### DISCUSSION TITLE
SR Health annotations used in a stereotype

### DISCUSSION THREAD

**Discussion by famod** [quarkusio/quarkus#36106] [Community]
Hi @xstefank, @jmartisk and all other experts in this area!

I'm trying to use `@Wellness` and `@HealthGroup` like this:
```java
    @WellnessCheck
    static class SomeWellnessCheck implements HealthCheck {
        // ...
    }

    @Singleton
    @Wellness
    @HealthGroup("my-group")
    @Retention(RUNTIME)
    @Stereotype
    @Target(TYPE)
    @interface WellnessCheck {
    }
```
but my checks are not picked up.

It does work when I add those annotations directly to each actual check.

Is this a bug or a missing feature or am I doing something wrong here?

Thanks!

**Comment by xstefank** [comment]
I would expect this to work since we pick the CDI beans with instance injection with Literal selections. @mkouba Should CDI stereotypes work in this case? - https://github.com/smallrye/smallrye-health/blob/main/implementation/src/main/java/io/smallrye/health/SmallRyeHealthReporter.java#L72-L74

**Comment by mkouba** [comment]
I think that it should work but we'll need a reproducer app to find out where the problem is.

**Reply by famod** [reply]
Done: #36187
